### ISSUE TITLE
Reduce CLI binary size

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-workflows#3576]
# Summary

The CLIs binaries are getting bigger and bigger. We should compact them.

# Motivation

So they are faster to download.

# Proposal

1. Strip debugging symbols.
2. Use UPX - subject to [LICENCE](https://github.com/upx/upx/blob/v3.96/LICENSE)

https://dev.to/akshaybharambe14/guide-to-compress-golang-binaries-2d95

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by alexec** [comment]
data:
```
-rwxr-xr-x  1 acollins8  staff    74M 23 Jul 15:08 argo-darwin-amd64
-rwxr-xr-x  1 acollins8  staff    57M 23 Jul 15:07 argo-linux-amd64
-rwxr-xr-x  1 acollins8  staff    54M 23 Jul 15:07 argo-linux-arm64
-rwxr-xr-x  1 acollins8  staff    55M 23 Jul 15:08 argo-linux-ppc64le
-rwxr-xr-x  1 acollins8  staff    57M 23 Jul 15:08 argo-linux-s390x
-rwxr-xr-x  1 acollins8  staff    57M 23 Jul 15:08 argo-windows-amd64
```

**Comment by alexec** [comment]
Sneak into sprint.

**Comment by alexec** [comment]
Why not just gzip the binaries?

**Comment by alexec** [comment]
gzip works well.

**Comment by alexec** [comment]
Available for testing in [v2.10.0-rc1](https://github.com/argoproj/argo/releases/tag/v2.11.0-rc1).
